src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AuthorisedGetCaseViewOperation.java

```diff
@@ -6,6 +6,7 @@
 import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
 import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_UPDATE;
 
+import org.apache.commons.lang.StringUtils;
```

Comment:
Fatih would it make sense to have a separate filter dedicated to role authorisation? for example RoleAuthorisedGetCaseViewOperation?
This class is mainly for CRUD, aren't we mixing separate things?

Reply:
I bet this code for checking the role authorisation is potentially highly reusable. And could be used for the other thickets too related to role based authorisation.
What if for example we have CaseView implement an interface called `RoleBased'. The we could have the `RoleBasedAuthorisation` operate on generic entities of type `RoleBased'. Which provide a method `getRole()`

Reply:
Role Based Tabs is yet another CRUD implementation for Tabs. Since Tabs are meaningful only for Case Views, it's R of CRUD we have here. I assume the name Role based may be misleading since whole CRUD implementation is about role based access to jurisdiction, case type, state or case fields. 
The current implementation is at the point where we're filtering tabs, so I feel it's somewhat a relevant/good place. I would say we keep the current implementation, finish this task and look for an optimisation/refactoring on the oncoming tasks.

Reply:
ok then. Let's keep an eye on avoiding potential duplication we might introduce in the next role related tickets tough